x-pack/metricbeat/module/meraki/network_health/networks.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package network_health
+
+import (
+	"strconv"
+
+	sdk "github.com/meraki/dashboard-api-go/v3/sdk"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/meraki"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// ID is the unique identifier for all networks
+type ID string
+
+// Network contains attributes, statuses and metrics for Meraki networks
+type Network struct {
+	id       ID
+	name     string
+	vpnPeers *[]sdk.ResponseItemApplianceGetOrganizationApplianceVpnStatsMerakiVpnpeers
+}
+
+func reportNetworkMetrics(reporter mb.ReporterV2, organizationID string, networks map[ID]*Network) {
+	metrics := []mapstr.M{}
+	for _, network := range networks {
+		if network == nil || network.vpnPeers == nil {
+			continue
+		}
+
+		metric := mapstr.M{
+			"network.id":   network.id,
+			"network.name": network.name,
+		}
+
+		peers := make([]mapstr.M, len(*network.vpnPeers))
+		for i, peer := range *network.vpnPeers {
+			peers[i] = vpnPeerToMapStr(&peer)
+		}
+		metric["network.vpn_peers"] = peers
+
+		metrics = append(metrics, metric)
+	}
+
+	meraki.ReportMetricsForOrganization(reporter, organizationID, metrics)
+}
+
+func vpnPeerToMapStr(peer *sdk.ResponseItemApplianceGetOrganizationApplianceVpnStatsMerakiVpnpeers) mapstr.M {
+	res := mapstr.M{
+		"network_id":   peer.NetworkID,
+		"network_name": peer.NetworkName,
+	}
+
+	if peer.UsageSummary != nil {
+		recv, err := strconv.Atoi(peer.UsageSummary.ReceivedInKilobytes)
+		sent, err := strconv.Atoi(peer.UsageSummary.SentInKilobytes)
+
+		if err == nil {
+			res["usage_summary"] = mapstr.M{
+				"received.kb": recv,
+				"sent.kb":     sent,
+			}
+		}
+
+	}
+
+	if peer.JitterSummaries != nil {
+		jitterSummaries := make([]mapstr.M, len(*peer.JitterSummaries))
+		for i, jitter := range *peer.JitterSummaries {
+			jitterSummaries[i] = mapstr.M{
+				"jitter_avg":      jitter.AvgJitter,
+				"jitter_max":      jitter.MaxJitter,
+				"jitter_min":      jitter.MinJitter,
```

Comment:
since these all go in the `jitter_summaries` key could we only keep the aggregate type? same question for `latency_summaries`, `loss_percentage_summaries` and `mos_summaries`
```suggestion
				"avg":      jitter.AvgJitter,
				"max":      jitter.MaxJitter,
				"min":      jitter.MinJitter,
```